ufl/algorithms/apply_derivatives.py

```diff
@@ -593,6 +591,25 @@ def _grad(self, o):
         # 1) n = count number of Grads, get f
         # 2) if not f.has_derivatives(n): return zero(...)
 
+    def _grad_to_reference_grad(self, o, K):
```

Comment:
Nitpick, this function doesn't need `self` I think, so could it just be a free function?

Reply:
Sure! I put it there because it is only used by the `GradRuleset` class, but I'm happy to move it